tests/extras/test_aggregate.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+# © H2O.ai 2018; -*- encoding: utf-8 -*-
+#   This Source Code Form is subject to the terms of the Mozilla Public
+#   License, v. 2.0. If a copy of the MPL was not distributed with this
+#   file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#-------------------------------------------------------------------------------
+#
+# Test creating a datatable from various sources
+#
+#-------------------------------------------------------------------------------
+import math
+import pytest
+import datatable as dt
+from datatable import ltype, stype, DatatableWarning
+import datatable.extras.aggregate
+from datatable import stype
+
+#-------------------------------------------------------------------------------
+# Aggregate 1D 
+#-------------------------------------------------------------------------------
+
+def test_aggregate_1d_continuous_integer_sorted():
+    d0 = dt.Frame([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    d1 = d0.aggregate(1e-15, 3)
+    assert d1.internal.check()
+    assert d1.shape == (10, 2)
+    assert d1.ltypes == (ltype.real, ltype.int)
+    assert d1.topython() == [[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], 
+                             [0, 0, 0, 0, 1, 1, 1, 2, 2, 2]]
+    
+
+def test_aggregate_1d_continuous_integer_random():
+    d0 = dt.Frame([9, 8, 2, 3, 3, 0, 5, 5, 8, 1])
+    d1 = d0.aggregate(1e-15, 3)
+    assert d1.internal.check()
+    assert d1.shape == (10, 2)
+    assert d1.ltypes == (ltype.real, ltype.int)
+    assert d1.topython() == [[9, 8, 2, 3, 3, 0, 5, 5, 8, 1], 
+                             [2, 2, 0, 0, 0, 0, 1, 1, 2, 0]]
+    
+    
+def test_aggregate_1d_continuous_real_sorted():
+    d0 = dt.Frame([0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9])
+    d1 = d0.aggregate(1e-15, 3)
+    assert d1.internal.check()
+    assert d1.shape == (10, 2)
+    assert d1.ltypes == (ltype.real, ltype.int)
+    assert d1.topython() == [[0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9], 
+                             [0, 0, 0, 0, 1, 1, 1, 2, 2, 2]]
+    
+     
+def test_aggregate_1d_continuous_real_random():
+    d0 = dt.Frame([0.7, 0.7, 0.5, 0.1, 0.0, 0.9, 0.1, 0.3, 0.4, 0.2])
+    d1 = d0.aggregate(1e-15, 3)
+    assert d1.internal.check()
+    assert d1.shape == (10, 2)
+    assert d1.ltypes == (ltype.real, ltype.int)
+    assert d1.topython() == [[0.7, 0.7, 0.5, 0.1, 0.0, 0.9, 0.1, 0.3, 0.4, 0.2], 
+                             [2, 2, 1, 0, 0, 2, 0, 0, 1, 0]]
+    
+#-------------------------------------------------------------------------------
+# Aggregate 2D 
+#-------------------------------------------------------------------------------    
+    
+def test_aggregate_2d_continuous_integer_sorted():
+    d0 = dt.Frame([[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], 
+                   [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]])
+    d1 = d0.aggregate(1e-15, 0, 3, 3)
```

Comment:
perhaps it's better to use named parameters here... How do you know what those `0, 3, 3` mean?

Reply:
The definition of aggregate is as follows:

`def aggregate(self, epsilon=1.0e-15, n_bins=500, nx_bins=50, ny_bins=50, max_dimensions=50, seed=0)`

So these are just the number of bins for the test.